File: system-test/dotnet/SystemTests/Legacy/Mod03/SmokeTests/External/ErpSmokeTest.cs
using SystemTests.Legacy.Mod03.Base;
using Xunit;

namespace SystemTests.Legacy.Mod03.SmokeTests.External;

public class ErpSmokeTest : BaseRawTest
{
    private const string HealthEndpoint = "/health";

    public override Task InitializeAsync()
    {
        SetUpExternalHttpClients();
        return Task.CompletedTask;
    }

    [Fact]
    public async Task ShouldBeAbleToGoToErp()
    {
        var uri = new Uri(_configuration.ErpBaseUrl + HealthEndpoint);
        var request = new HttpRequestMessage(HttpMethod.Get, uri);

        var response = await _erpHttpClient!.SendAsync(request);

        Assert.Equal(200, (int)response.StatusCode);
    }
}

File: system-test/java/src/test/java/com/optivem/eshop/systemtest/legacy/mod02/smoke/external/ErpSmokeTest.java
package com.optivem.eshop.systemtest.legacy.mod02.smoke.external;

import com.optivem.eshop.systemtest.legacy.mod02.base.BaseRawTest;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.net.URI;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

import static org.assertj.core.api.Assertions.assertThat;

class ErpSmokeTest extends BaseRawTest {
    private static final String HEALTH_ENDPOINT = "/health";

    @BeforeEach
    void setUp() {
        setUpExternalHttpClients();
    }

    @Test
    void shouldBeAbleToGoToErp() throws Exception {
        var uri = URI.create(getErpBaseUrl() + HEALTH_ENDPOINT);
        var request = HttpRequest.newBuilder()
                .uri(uri)
                .GET()
                .build();

        var response = erpHttpClient.send(request, HttpResponse.BodyHandlers.ofString());

        assertThat(response.statusCode()).isEqualTo(200);
    }
}


